### PR TITLE
Telink B91 IEEE802154 driver support

### PR DESCRIPTION
**Pull request by yurvyn**
IEEE802154 driver basic support for Telink B91 platform.

Signed-off-by: Yuriy Vynnychek <yura.vynnychek@telink-semi.com>

### CODE REVIEW

**Comment by galak** [comment]
need a devicetree binding / support for this.

**Comment by yurvyn** [comment]
> need a devicetree binding / support for this.

As I see almost all other vendors' drivers are not using devicetree binding / support.
Only 5 drivers are using devicetree, but all of them are related to "spi-device" and this is not a case for our B91 IEEE802154 driver.

@galak,
Please confirm.

File: boards/riscv/tlsr9518adk80d/doc/index.rst [hunk 1]
```diff
@@ -63,6 +63,8 @@ The Zephyr TLSR9518ADK80D board configuration supports the following hardware fe
 +----------------+------------+------------------------------+
 | UART           | on-chip    | serial                       |
 +----------------+------------+------------------------------+
+| RADIO          | on-chip    | ieee802154, OpenThred        |
```
Comment (rlubos) [inline boards/riscv/tlsr9518adk80d/doc/index.rst:66]:
```suggestion
| RADIO          | on-chip    | ieee802154, OpenThread        |
```

File: boards/riscv/tlsr9518adk80d/doc/index.rst [hunk 2]
```diff
@@ -77,6 +79,8 @@ The following example projects are supported:
 - samples/subsys/console/getline
 - samples/subsys/shell/shell_module
 - samples/subsys/cpp/cpp_synchronization
+- samples/net/sockets/echo_client (OpenThread and IEEE80154)
+- samples/net/sockets/echo_server (OpenThread and IEEE80154)
```
Comment (rlubos) [inline boards/riscv/tlsr9518adk80d/doc/index.rst:83]:
```suggestion
- samples/net/sockets/echo_client (OpenThread and IEEE802154)
- samples/net/sockets/echo_server (OpenThread and IEEE802154)
```

File: drivers/ieee802154/ieee802154_b91.h [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	RF_POWER_P4p35dBm,      /*   4.3 dBm:   4 */
+	RF_POWER_P5p68dBm,      /*   5.6 dBm:   5 */
+	RF_POWER_P5p68dBm,      /*   5.6 dBm:   6 */
+	RF_POWER_P6p98dBm,      /*   6.9 dBm:   7 */
+	RF_POWER_P8p05dBm,      /*   8.0 dBm:   8 */
+	RF_POWER_P9p11dBm,      /*   9.1 dBm:   9 */
+};
+
+/* data structure */
+struct b91_data_t {
```
Comment (rlubos) [inline drivers/ieee802154/ieee802154_b91.h:93]: No need for `_t` suffix.

**Comment by galak** [comment]
> As I see almost all other vendors' drivers are not using devicetree binding / support.
> Only 5 drivers are using devicetree, but all of them are related to "spi-device" and this is not a case for our B91 IEEE802154 driver.
> 
> @galak,
> Please confirm.

Those drivers have been in tree for some time and need to get converted at some point to use devicetree.  As this is a new driver it should utilize devicetree from the start.

**Comment by yurvyn** [comment]
> need a devicetree binding / support for this.

Done. 
Please review.

File: drivers/ieee802154/ieee802154_b91.h (hunk 3 above)
Comment (yurvyn) [reply drivers/ieee802154/ieee802154_b91.h]: Done.

File: boards/riscv/tlsr9518adk80d/doc/index.rst (hunk 2 above)
Comment (yurvyn) [reply boards/riscv/tlsr9518adk80d/doc/index.rst]: Done.

File: boards/riscv/tlsr9518adk80d/doc/index.rst (hunk 1 above)
Comment (yurvyn) [reply boards/riscv/tlsr9518adk80d/doc/index.rst]: Done.

**Review by rlubos — APPROVED** [review]

**Comment by yurvyn** [comment]
One more approval is required.
Could anyone please approve the PR?

**Review by cfriedt — APPROVED** [review]

**Comment by cfriedt** [comment]
@yurvyn - can you rebase and resolve conflicts please?

**Comment by cfriedt** [comment]
@yurvyn - can you rebase and resolve conflicts?

**Review by tbursztyka — APPROVED** [review]
looks pretty clean

**Comment by nashif** [comment]
@yurvyn  needs a rebase

**Pull request merged**